### DISCUSSION TITLE
Relative link in monorepo

### DISCUSSION THREAD

**Discussion by BennyEvans** [facebook/docusaurus#6814] [Q&A]
Hi,

I currently have a site setup as a monorepo such as that below:

```
/repo
    /website <- Existing website
    /docs  <- Docusaurus in docs only mode
```
The project is built into a single build directory and works well, however, I was wondering if there is a way to specify a relative link in docusarus without the baseURL being applied. My current configuration sets `baseUrl: '/docs/'` as the docusaurus docs are served from /docs/ on my website. In the docusaurus config, I'd like to be able to set options like `favicon: '/img/favicon.ico'` to point to my websites `/img/favicon.ico` and not `/docs/img/favicon.ico`.  Currently after building, the baseUrl would be applied and I cannot specify `/img/favicon.ico`  as the favicon address. Similarly, I'd like my docs navbar title to navigate back to the main website using relative URLs. Is there any way to specify a relative URL in docusaurus and tell it to not add the baseUrl?

If this isn't currently possible, could this be a feature? Perhaps starting a URL with an ! or some kind of symbol could tell docusaurus not to use baseUrl in this circumstance.

Cheers,
Ben

**Comment by Josh-Cena** [comment]
Indeed, I don't think there's a way to tell Docusaurus to escape the base URL, but it would be useful! As of now, you need to hardcode the entire URL, like `favicon: 'https://yoursite.com/img/favicon.ico'`

**Reply by miller-productions** [reply]
Hi there, 

Just dropping in. I started investigating Docusaurus for the first time yesterday and this was the very first question that came to mind. i.e. it would be interesting to figure out how to get Docusaurus to be be able to support having a single source of truth for common static assets such as logos, favicons, etc. that can be shared across various packages in a larger project ecosystem.

Similar to what @BennyEvans has said, say you have a single `logo.svg` for you project and you wish to use that in several packages for your project....

```
/repo
  /assets
    /img
      logo.svg    <--- a common file that several packages would want to reference/use
  /packages
    /web-app
    /docs-site   <--- this is the Docusaurus source
   /etc...
```

**Reply by slorber** [reply]
@miller-productions that looks like a different case here.

In @BennyEvans case, he knows that the `https://yoursite.com/img/favicon.ico` favicon exists and is already hosted.

In your case, it's unclear who is responsible for hosting your `/repo/assets/img/logo.svg` file and what its final url should be.

If you want Docusaurus to host the favicon:
- You can also add a `cp ./repo/assets/img/logo.svg ./packages/docs-site/static` cli command as part of your build process
- You can create a Docusaurus plugin that adds some webpack config like the CopyWebpackPlugin to copy the favicon to the appropriate place (somehow the same in the end)

If you don't want Docusaurus to host your favicon, then you are on your own and you should figure out how to host the favicon yourself, and use an absolute URL as suggested above.
